trin-core/src/portalnet/overlay.rs

```diff
@@ -138,9 +143,111 @@ impl OverlayProtocol {
             data_radius: Arc::new(RwLock::new(data_radius)),
             kbuckets,
             db,
+            utp_listener,
         }
     }
 
+    pub async fn process_response(
+        &self,
+        response_vec: Vec<u8>,
+        enr: Enr,
+        content_keys_vec: Vec<Vec<u8>>,
+    ) -> Result<(), String> {
+        let response = match Message::from_bytes(&response_vec) {
+            Ok(Message::Response(r)) => r,
+            Ok(_) => return Err("Invalid message".to_owned()),
+            Err(e) => return Err(format!("Invalid response: {}", e)),
+        };
+
+        match response {
+            Response::Pong(Pong { .. }) => {
+                // handle(PONG)
+            }
+            Response::Nodes(Nodes { .. }) => {
+                // handle(NODES)
+            }
+            Response::FoundContent(FoundContent { .. }) => {
+                // handle(NODES)
+            }
+            // might not be a good idea to make a separate function because of what is needed
+            Response::Accept(Accept {
+                connection_id,
+                content_keys,
+            }) => {
+                let (tx, mut rx) = mpsc::unbounded_channel::<String>();
+
+                // Add connection_id we will be receiving to the listening list and the
+                // response number. The connection_id is + 1 because that is their id
```

Comment:
Even with the comment, I don't understand why we add 1 here.

Reply:
We aren't going to actually use this till we are done testing, I have talked with some of the Fluffy team. I also have to double-check if this is the one that should be plus one.

Anyways we will probably add both the conn_recv_id and the conn_send_id. The conn_recv_id is used on the first packet for the handshake. After the first packet in the handshake, they both use conn_send_id or their conn_send_id.

One will be receiving packets with + 1 as each node has a different conn_send_id with a 1 increment difference

Reply:
This confused me as well until I read the uTP spec where they describe connection_id:

> Each socket has one connection ID for sending packets and a different connection ID for receiving packets. The endpoint initiating the connection decides which ID to use, and the return path has the same ID + 1.

I think what Kolby is trying to say is that node A (the offerer) receives the `connection_id` from node B (the acceptor) and uses this as its `conn_send_id`. For receiving uTP packets, it will listen on a socket with `id+1`, ie it's `conn_recv_id`.

Reply:
Looking over the code again, I think there is some difference between original uTP spec and current implementation. I think there is some misalignment in what we are using as id for send/recv. ie, a node can have recv id as just `id` and send as `id+1`, or it can have the reverse, it doesn't really matter, but we just have to make sure it is consistent.